### PR TITLE
nixos/profiles: optimize base and minimal profiles

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Small optimize base profile:
- nixos/profiles/base: remove duplicate systemPackages
- nixos/profiles/base: remove duplicate and optimize fsPackages
- ~~nixos/profiles/base: remove duplicate systemPackages~~
- nixos/modules/installer/netboot/netboot-minimal: reduce closure size

Allows you to eliminate the installation of some packages when using a minimal set of supported fileSystems, example:
```
...
  netboot = import (pkgs.path + "/nixos/lib/eval-config.nix") {
      modules = [
        (pkgs.path + "/nixos/modules/installer/netboot/netboot-minimal.nix")
        module
      ];
    };
    module = {
      boot = {
        supportedFilesystems = lib.mkForce [ "vfat" "ntfs" "zfs" ];
...
```
The parameter `boot.initrd.supportedfileSystems = [" vfat "];` in `nixos/modules/installer/cd-dvd/iso-image.nix` is duplicated in `nixos/modules/profiles/base.nix.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
> I suppose we could have a base-minimal and a base and base-minimal can be used for netboot-minimal?

Checked working with test `nixos/tests/kexec.nix` and loading `netboot-minimal` image from grub menu on home server.

**Comment by Izorkin** [comment]
It was possible to significantly reduce the size of the image while maintaining the necessary functionality:
```
l /boot | grep rescue
-r--r--r--  1 root root 392M Jan  1  1970 rescue-initrd
-r--r--r--  1 root root 7.5M Jan  1  1970 rescue-kernel
```

**Comment by RaitoBezarius** [comment]
> It was possible to significantly reduce the size of the image while maintaining the necessary functionality:
> 
> ```
> l /boot | grep rescue
> -r--r--r--  1 root root 392M Jan  1  1970 rescue-initrd
> -r--r--r--  1 root root 7.5M Jan  1  1970 rescue-kernel
> ```

Awesome! What do you think could be areas of further improvements? Are those initrds compressed BTW? (can you try zstd or xz compression?)

**Comment by Izorkin** [comment]
Used is maximum compression - https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-squashfs.nix#L8

**Comment by RaitoBezarius** [comment]
> Used is maximum compression - https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-squashfs.nix#L8

Do you have the numbers of `systemd-analyze blame` when you boot?

**Comment by Izorkin** [comment]
Before PR:
```
838ms systemd-modules-load.service
642ms dev-loop0.device
477ms user@1737.service
385ms dbus.service
273ms resolvconf.service
246ms systemd-tmpfiles-setup-dev.service
234ms systemd-oomd.service
229ms systemd-udevd.service
208ms zfs-mount.service
184ms mount-pstore.service
177ms systemd-logind.service
161ms systemd-udev-trigger.service
132ms sshd@0-192.168...-192.168....service
121ms systemd-timesyncd.service
107ms systemd-journald.service
102ms network-setup.service
100ms network-addresses-enp1s0.service
 91ms dev-hugepages.mount
 91ms modprobe@drm.service
 90ms dev-mqueue.mount
 88ms audit.service
 73ms systemd-remount-fs.service
 64ms sys-fs-fuse-connections.mount
 63ms sys-kernel-config.mount
 47ms systemd-journal-catalog-update.service
 46ms systemd-journal-flush.service
 44ms modprobe@efi_pstore.service
 36ms systemd-random-seed.service
 30ms systemd-update-utmp.service
 29ms user-runtime-dir@1737.service
 28ms systemd-tmpfiles-setup.service
 27ms systemd-user-sessions.service
 20ms kmod-static-nodes.service
 19ms modprobe@configfs.service
 16ms systemd-sysctl.service
 15ms modprobe@fuse.service
 15ms network-local-commands.service
 13ms nscd.service
  8ms zfs-share.service
  7ms systemd-update-done.service
```

After PR:
```
873ms systemd-modules-load.service
711ms dev-loop0.device
424ms user@1737.service
316ms systemd-udevd.service
289ms dbus.service
258ms systemd-oomd.service
239ms resolvconf.service
208ms systemd-tmpfiles-setup-dev.service
180ms zfs-mount.service
158ms mount-pstore.service
148ms systemd-timesyncd.service
140ms systemd-udev-trigger.service
134ms systemd-journald.service
133ms sshd@0-192.168....-192.168.....service
110ms systemd-logind.service
105ms network-setup.service
105ms network-addresses-enp1s0.service
 87ms dev-hugepages.mount
 87ms modprobe@drm.service
 86ms audit.service
 86ms dev-mqueue.mount
 68ms systemd-remount-fs.service
 58ms sys-fs-fuse-connections.mount
 56ms sys-kernel-config.mount
 49ms systemd-random-seed.service
 45ms systemd-journal-flush.service
 43ms systemd-journal-catalog-update.service
 38ms systemd-tmpfiles-setup.service
 31ms network-local-commands.service
 29ms systemd-update-utmp.service
 28ms user-runtime-dir@1737.service
 26ms systemd-user-sessions.service
 19ms kmod-static-nodes.service
 19ms modprobe@configfs.service
 19ms systemd-sysctl.service
 16ms modprobe@fuse.service
 16ms modprobe@efi_pstore.service
 10ms zfs-share.service
  7ms nscd.service
  7ms systemd-update-done.service
```

Data collected on a test virtual machine.

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 1]
```diff
@@ -808,11 +808,6 @@ in
         touch /etc/NIXOS
         ${config.nix.package.out}/bin/nix-env -p /nix/var/nix/profiles/system --set /run/current-system
       '';
-
-    # Add vfat support to the initrd to enable people to copy the
-    # contents of the CD to a bootable USB stick.
-    boot.initrd.supportedFilesystems = [ "vfat" ];
```
Comment (lheckemann) [inline nixos/modules/installer/cd-dvd/iso-image.nix:814]: We should keep this. While all the images produced by nixpkgs get vfat via `profile/base.nix`, external consumers can (and do) still pull in `iso-image.nix` without `base.nix` and it would be good to continue supporting the USB use case for such images.

File: nixos/modules/tasks/filesystems/vfat.nix [hunk 2]
```diff
@@ -11,7 +11,7 @@ in
 {
   config = mkIf (any (fs: fs == "vfat") config.boot.supportedFilesystems) {
 
-    system.fsPackages = [ pkgs.dosfstools ];
+    system.fsPackages = [ pkgs.dosfstools pkgs.mtools ];
```
Comment (lheckemann) [inline nixos/modules/tasks/filesystems/vfat.nix:14]: mtools seems pretty obscure and unnecessary for supporting access to FAT filesystems. Why is this needed?

Comment (Izorkin) [reply nixos/modules/tasks/filesystems/vfat.nix:14]: The `mtools` utility is used when working with the FAT file system.
See - https://github.com/NixOS/nixpkgs/pull/146288

File: nixos/modules/installer/cd-dvd/iso-image.nix (hunk 1 above)
Comment (Izorkin) [reply nixos/modules/installer/cd-dvd/iso-image.nix]: Fixed.

**Review by RaitoBezarius — APPROVED** [review]
LGTM

**Review by SebTM — APPROVED** [review]
